Added ability to pass username and password for Registry Authentication

philippreston opened the pull request:
This adds the ability to pass through username/password/email to the registryCredentials block in docker-java.  This feature is configured via the following:

Param | Description | Type | Default
-------|------------|-------|---------
dockerRepoAuth | Enable registryCredentials | boolean | false
dockerRepoUsername | The registry username | String | null
dockerRepoPassword | The registry password | String | null
dockerRepoEmail | The registry email | String | null

This is for use with our internal docker repo.

Many Thanks

philippreston commented:
Hey 

Just checking in on whether this is something that would be acceptable to merge in?  The plugin (including this change) is something I am keen to use in an active project.

Many thanks

Phil

liviutudor commented:
Hi Phil,

Just returning from a holiday, and looking into this today.

Thanks for your pull request.

liviutudor commented on src/functionalTest/groovy/nebula/plugin/docker/NebulaDockerPluginTest.groovy:
```diff
@@ -261,4 +261,37 @@ class NebulaDockerPluginTest extends ProjectSpec {
         //TODO: how to check the list of classpath?
     }
+
+    def "enabling repoAuth and setting the username password and email on docker.registryCredentials"() {
```
can you also add a test to ensure that flipping the switch on `dockerRepoAuth` does NOT include the credentials?

liviutudor commented on src/test/groovy/nebula/plugin/docker/NebulaDockerSensibleDefaultsTest.groovy:
```diff
@@ -49,5 +49,9 @@ class NebulaDockerSensibleDefaultsTest extends Specification {
         ext.dockerFile == x.DEF_DOCKER_FILE
         ext.appDirLatest == "/${appName}-latest"
         ext.appDir == "/${appName}-1.2.3"
+        !ext.dockerRepoAuth
```
+1 for thinking to add these here.

philippreston commented:
Hey Liviu

Thanks for that.  I have made the change and added an additional unit test for the flip on the dockerRepoAuth.

I've squashed the additional commit.

Many thanks for looking at this

Phil
